rs/tests/system_tests.bzl

```diff
@@ -341,73 +241,104 @@ def system_test(
+        srcs = ["//rs/tests:run_systest.sh"],
+        data = data + [test_driver_target],
+        env = env | {
+            "RUN_SCRIPT_DRIVER_EXTRA_ARGS": " ".join(extra_args_simple),
+            "RUN_SCRIPT_TEST_EXECUTABLE": "$(rootpath {})".format(test_driver_target),
+        },
         env_inherit = env_inherit,
-        tags = tags + ["requires-network", "system_test"] +
-               (["manual"] if "colocate" in tags else []),
+        tags = tags + (["manual"] if "colocate" in tags else []),
         target_compatible_with = ["@platforms//os:linux"],
         timeout = test_timeout,
         flaky = flaky,
-        exclude_logs = exclude_logs,
+        visibility = visibility,
     )
 
-    env = env | {
-        "COLOCATED_TEST": test_name,
-        "COLOCATED_TEST_DRIVER_VM_REQUIRED_HOST_FEATURES": json.encode(colocated_test_driver_vm_required_host_features),
-        "COLOCATED_TEST_DRIVER_VM_RESOURCES": json.encode(colocated_test_driver_vm_resources),
-    }
-
-    if colocated_test_driver_vm_enable_ipv4:
-        env.update({"COLOCATED_TEST_DRIVER_VM_ENABLE_IPV4": "1"})
-
-    if colocated_test_driver_vm_forward_ssh_agent:
-        env.update({"COLOCATED_TEST_DRIVER_VM_FORWARD_SSH_AGENT": "1"})
-
-    visibility = kwargs.get("visibility", ["//visibility:public"])
-
-    run_system_test(
+    sh_test(
+        srcs = ["//rs/tests:run_systest.sh"],
         name = test_name + "_colocate",
-        src = "//rs/tests/idx:colocate_test_bin",
-        colocated_test_bin = test_driver_target,
-        env_var_files = env_var_files,
-        runtime_deps = deps + [
+        data = data + [
             "//rs/tests:colocate_uvm_config_image",
+            "//rs/tests/idx:colocate_test_bin",
             test_driver_target,
         ],
         env_inherit = env_inherit,
-        env = env,
-        icos_images = icos_images,
-        tags = tags + ["requires-network", "system_test"] +
-               (["colocated"] if "colocate" in tags else ["manual"]) +
-               additional_colocate_tags,
+        env = env | {
+                  "COLOCATED_TEST_BIN": "$(rootpath {})".format(test_driver_target),
+                  "RUN_SCRIPT_TEST_EXECUTABLE": "$(rootpath //rs/tests/idx:colocate_test_bin)",
+                  "RUN_SCRIPT_DRIVER_EXTRA_ARGS": " ".join(extra_args_colocated),
+                  "COLOCATED_TEST": test_name,
+                  "COLOCATED_TEST_DRIVER_VM_REQUIRED_HOST_FEATURES": json.encode(colocated_test_driver_vm_required_host_features),
+                  "COLOCATED_TEST_DRIVER_VM_RESOURCES": json.encode(colocated_test_driver_vm_resources),
+              } | ({"COLOCATED_TEST_DRIVER_VM_ENABLE_IPV4": "1"} if colocated_test_driver_vm_enable_ipv4 else {}) |
+              ({"COLOCATED_TEST_DRIVER_VM_FORWARD_SSH_AGENT": "1"} if colocated_test_driver_vm_forward_ssh_agent else {}),
+        tags = tags + (["colocated"] if "colocate" in tags else ["manual"]) + additional_colocate_tags,
```

Comment:
Those tags are a little surprising, though this seems to be the intended behavior